types/ejson/ejson-tests.ts

```diff
@@ -1,70 +1,102 @@
+    CustomType,
+    equals as importedEquals,
+    fromJSONValue as importedFromJSONValue,
+    isBinary as importedIsBinary,
+    newBinary as importedNewBinary,
+    parse as importedParse,
+    stringify as importedStringify,
+    toJSONValue as importedToJSONValue,
+} from 'ejson';
 
 function testImportedClone() {
-  var obj: Object = {
-    a: "a"
-  };
-  var retval: Object = importedClone(obj);
+    var obj: Object = {
+        a: 'a',
+    };
+    var retval: Object = importedClone(obj);
 
-  var str: string = "as";
-  var retval2: string = importedClone(str);
+    var str: string = 'as';
+    var retval2: string = importedClone(str);
 }
 
 function testParse() {
-  var str: string = '{a:"a"}';
-  importedParse(str);
+    var str: string = '{a:"a"}';
+    importedParse(str);
 }
 
 function testStringify() {
-  var obj: any = {a:"a"};
-  var retval: string = importedStringify(obj);
+    var obj: any = { a: 'a' };
+    var retval: string = importedStringify(obj);
 }
 
 function testStringifyIndent() {
-  var obj: any = {a: "a"};
-  var retval: string = importedStringify(obj, { indent: true });
+    var obj: any = { a: 'a' };
+    var retval: string = importedStringify(obj, { indent: true });
 }
 
 function testStringifyCanonical() {
-  var obj: any = {a: "a"};
-  var retval: string = importedStringify(obj, { canonical: true });
+    var obj: any = { a: 'a' };
+    var retval: string = importedStringify(obj, { canonical: true });
 }
 
 function testStringifyOptions() {
-  var obj: any = {a: "a"};
-  var retval: string = importedStringify(obj, { canonical: true, indent: 'hello' });
+    var obj: any = { a: 'a' };
+    var retval: string = importedStringify(obj, { canonical: true, indent: 'hello' });
 }
 
 function testToJSONValue() {
-  var obj: any = {a:"a"};
-  var retval: string = importedToJSONValue(obj);
+    var obj: any = { a: 'a' };
+    var retval: string = importedToJSONValue(obj);
 }
 
 function testFromJSONValue() {
-  var str: string = '{a:"a"}';
-  importedFromJSONValue(str);
+    var str: string = '{a:"a"}';
+    importedFromJSONValue(str);
 }
 
 function testIsBinary() {
-  var val: any = 'sasda';
-  var retval: boolean = importedIsBinary(val);
+    var val: any = 'sasda';
+    var retval: boolean = importedIsBinary(val);
 }
 
 function testNewBinary() {
-  var retval: Uint8Array = importedNewBinary(3);
+    var retval: Uint8Array = importedNewBinary(3);
 }
 
 function testEquals() {
-  var a: any;
-  var b: any;
-  var retval: boolean = importedEquals(a,b);
+    var a: any;
+    var b: any;
+    var retval: boolean = importedEquals(a, b);
+}
+
+class MyInt implements CustomType {
+    constructor(public value: number) {}
+
+    typeName() {
+        return 'MyInt';
+    }
+
+    toJSONValue() {
+        return this.value.toString();
+    }
+
+    static fromJSONValue(jsonValue: unknown) {
+        const value = parseInt(jsonValue as string, 10);
+        return new MyInt(value);
+    }
+
+    // TODO doesn't compile
```

Comment:
Can you elaborate on your intent with these TODOs? It doesn't seem required to test the function you've added.

Reply:
They correspond with the commented-out fields on `CustomType` interface. I don't know why they don't compile, but it would be nice if they did. It's not essential, since the fields are optional, so I'm just leaving a note that it would be a good thing to figure out in future.

Reply:
I see. I'm assuming this has to do with the fact that the parameters have mismatching types? I would think that your comparison functions need to take the super type (`CustomType`), not the specific type, otherwise it wouldn't be comparable to other subtypes of `CustomType`. This is similar to Java/C# comparison methods.

Reply:
Thanks, fixed & removed TODO!